src/main/lrsql/admin/interceptors/reaction.clj

```diff
@@ -0,0 +1,174 @@
+        (if-some [err (s/explain-data rs/update-reaction-params-spec params)]
+          ;; Invalid parameters - Bad Request
+          (assoc (chain/terminate ctx)
+                 :response
+                 {:status 400
+                  :body   {:error (format "Invalid parameters:\n%s"
+                                          (-> err s/explain-out with-out-str))}})
+          ;; Valid params - continue
+          (assoc ctx ::data params))))}))
+
+(def validate-delete-reaction-params
+  "Validate valid params for reaction delete."
+  (interceptor
+   {:name ::validate-delete-reaction-params
+    :enter
+    (fn validate-params [ctx]
+      (let [params (-> (get-in ctx [:request :json-params])
+                       ru/json->input
+                       (update :reaction-id u/str->uuid))]
+        (if-some [err (s/explain-data rs/delete-reaction-params-spec params)]
+          ;; Invalid parameters - Bad Request
+          (assoc (chain/terminate ctx)
+                 :response
+                 {:status 400
+                  :body   {:error (format "Invalid parameters:\n%s"
+                                          (-> err s/explain-out with-out-str))}})
+          ;; Valid params - continue
+          (assoc ctx ::data params))))}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Terminal Interceptors
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(def create-reaction
+  "Create a new reaction and store it."
+  (interceptor
+   {:name ::create-reaction
+    :enter
+    (fn create-reaction [ctx]
+      (let [{lrs                            :com.yetanalytics/lrs
+             {:keys [title ruleset active]} ::data}
+            ctx
+            {:keys [result]}
+            (adp/-create-reaction lrs title ruleset active)]
+        (cond
+          (uuid? result)
+          (assoc ctx
+                 :response
+                 {:status 200 :body {:reactionId result}})
+          (= :lrsql.reaction/title-conflict-error result)
+          (assoc (chain/terminate ctx)
+                 :response
+                 {:status 400
+                  :body   {:error (format "Title \"%s\" is already in use."
+                                          title)}}))))}))
+
+(def get-all-reactions
+  "List all reactions."
+  (interceptor
+   {:name ::get-all-reactions
+    :enter
+    (fn get-all-reactions [ctx]
+      (let [{lrs :com.yetanalytics/lrs} ctx
+            result
+            (adp/-get-all-reactions lrs)]
+        (assoc ctx
+               :response
+               {:status 200 :body {:reactions
+                                   (map
+                                    (fn [reaction-record]
+                                      (-> reaction-record
+                                          (update :created u/time->str)
+                                          (update :modified u/time->str)))
```

Comment:
These calls to `u/time->str` should go into the `query-all-reactions` domain function (see `lrsql.ops.query.document/query-document` for an example

Reply:
c64eceadd8aeb018c6d1b8c622a410e0b5f6a3c5 had to split the spec tho to limit the footprint

Reply:
Here's a neat trick: since Clojure 1.11 you can use `:as-alias` to create keyword ns aliases.

Reply:
Hey that's cool (and I already know it) but in this case I'd like to keep the long-ish keywords thanks